Code/Source/URDF/RobotImporter/URDFPrefabMaker.cpp

```diff
@@ -174,36 +177,100 @@ namespace ROS2
+                    auto boxGeometry = std::dynamic_pointer_cast<urdf::Box>(geometry);
+                    entity->CreateComponent(LmbrCentral::EditorBoxShapeComponentTypeId);
+                    AZ::Vector3 boxDimensions = URDF::TypeConversions::ConvertVector3(boxGeometry->dim);
+                    entity->Activate();
+                    LmbrCentral::BoxShapeComponentRequestsBus::Event(entityId,
+                        &LmbrCentral::BoxShapeComponentRequests::SetBoxDimensions,
+                        boxDimensions);
+                    entity->Deactivate();
+                }
+                break;
+            case urdf::Geometry::MESH:
+                {
+                    auto meshGeometry = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+
+                    // TODO - a PoC solution, replace with something generic, robust, proper
+                    std::filesystem::path modelPath(m_modelFilePath.c_str());
+                    modelPath = modelPath.remove_filename();
+                    auto relativePathToMesh = std::regex_replace(meshGeometry->filename, std::regex("package://"), "");
+                    modelPath += relativePathToMesh;
+                    AZ_Warning("AddVisuals", false, "Adding mesh urdf_uri: %s, changed to filename: %s",
+                               meshGeometry->filename.c_str(), modelPath.c_str());
+
+                    // TODO load mesh, use asset processor
+                    entity->CreateComponent(AZ::Render::EditorMeshComponentTypeId);
+                    entity->Activate();
+                    AZ::Render::MeshComponentRequestBus::Event(entityId,
+                        &AZ::Render::MeshComponentRequestBus::Events::SetModelAssetPath,
+                        modelPath.c_str());
+                    entity->Deactivate();
+                    // TODO apply scale
+                }
+                break;
+            default:
+                AZ_Warning("AddVisuals", false, "Unsupported visual geometry type, %d", geometry->type);
+                return;
         }
+
         // TODO handle material
     }
 
     void URDFPrefabMaker::AddColliders(urdf::LinkSharedPtr /*link*/, AZ::EntityId /*entityId*/)
     {   // TODO - implement
     }
 
-    void URDFPrefabMaker::AddInertia(urdf::LinkSharedPtr /*link*/, AZ::EntityId /*entityId*/)
-    {   // TODO - implement
+    void URDFPrefabMaker::AddInertia(urdf::LinkSharedPtr link, AZ::EntityId entityId)
+    {
+        urdf::InertialSharedPtr inertial = link->inertial;
+        if (!inertial)
+        {   // it is ok not to have inertia in a link
+            return;
+        }
+
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        // TODO - consider explicit 2 arg constructor instead
+        PhysX::EditorRigidBodyConfiguration rigidBodyConfiguration;
+        rigidBodyConfiguration.m_mass = inertial->mass;
+        // TODO - is the origin.rotation part applicable? Does non-zero make value sense? Investigate.
```

Comment:
Answers to your question: 

> is the origin.rotation part applicable?

No

> Does non-zero make value sense? 

Yes

Reply:
I meant non zero for origin.rotation - which seems it doesn't make sense, not being applicable